src/coreComponents/fileIO/timeHistory/PackCollection.cpp

```diff
@@ -8,143 +8,211 @@ PackCollection::PackCollection ( string const & name, Group * parent )
   , m_objectPath( )
   , m_fieldName( )
   , m_setNames( )
-  , m_minimumSetSize( )
+  , m_setChanged( false )
+  , m_onlyOnSetChange( 0 )
+  , m_disableCoordCollection( false )
+  , m_initialized( false )
 {
-  registerWrapper( PackCollection::viewKeysStruct::objectPath, &m_objectPath ).
+  registerWrapper( PackCollection::viewKeysStruct::objectPathString(), &m_objectPath ).
     setInputFlag( InputFlags::REQUIRED ).
     setDescription( "The name of the object from which to retrieve field values." );
 
-  registerWrapper( PackCollection::viewKeysStruct::fieldName, &m_fieldName ).
+  registerWrapper( PackCollection::viewKeysStruct::fieldNameString(), &m_fieldName ).
     setInputFlag( InputFlags::REQUIRED ).
     setDescription( "The name of the (packable) field associated with the specified object to retrieve data from" );
 
-  registerWrapper( PackCollection::viewKeysStruct::setNames, &m_setNames ).
+  registerWrapper( PackCollection::viewKeysStruct::setNamesString(), &m_setNames ).
     setInputFlag( InputFlags::OPTIONAL ).
     setDescription( "The set(s) for which to retrieve data." );
 
-  registerWrapper( PackCollection::viewKeysStruct::minSetSize, &m_minimumSetSize ).
+  registerWrapper( PackCollection::viewKeysStruct::onlyOnSetChangeString(), &m_onlyOnSetChange ).
     setInputFlag( InputFlags::OPTIONAL ).
-    setDefaultValue( -1 ).
-    setDescription( "The minimum size of the set(s) to be collected (use for sets that expand during the simulation)." );
+    setDefaultValue( 0 ).
+    setDescription( "Whether or not to only collect when the collected sets of indices change in any way." );
 }
 
-void PackCollection::initializePostSubGroups()
+void PackCollection::initializePostSubGroups( )
 {
-  localIndex numSets = m_setNames.size( );
-  m_collectionCount = numSets == 0 ? 1 : numSets;
-  DomainPartition & domain = this->getGroupByPath< DomainPartition >( "/Problem/domain" );
-  updateSetsIndices( domain );
-  HistoryCollection::initializePostSubGroups();
+  if( !m_initialized )
+  {
+    DomainPartition & domain = this->getGroupByPath< DomainPartition >( "/Problem/domain" );
```

Comment:
@rrsettgast is this preferred over `getGlobalState().problemManager().domain()`?

Reply:
I would think to prefer `getGlobalState().problemManager().domain()`, which is what I was using originally, this came in via a merge and I left it.

Reply:
It might be the way it is to avoid dependencies.

Reply:
We have to fix this still...I was avoiding calls to `getGlobalState()` due to dependency issues, but I hadn't provided an accessor for domain. It isn't pleasant as it stands.